int overflows when compiling on arm32

I tried to compile juicefs 1.1.2 on arm32 platforms; I tried on two systems:
- first an an armv5tel processor and go1.21.5 and gcc 10.2.1, 
-  second with an armvh i.e armv7l processor,  the same version of go 1.21.5 and gcc 12.2.0.

I got exactly the same result.

First tkiv, don't compile. So I tried first to compile juicefs.lite, which do not include extra libraries.

The result is 

```
pkg/object/eos.go:47:29: cannot use 5 << 30 (untyped int constant 5368709120) as int value in struct literal (overflows)
pkg/object/ks3.go:66:29: cannot use 5 << 30 (untyped int constant 5368709120) as int value in struct literal (overflows)
pkg/object/minio.go:49:29: cannot use 5 << 30 (untyped int constant 5368709120) as int value in struct literal (overflows)
pkg/object/qingstor.go:55:29: cannot use 5 << 30 (untyped int constant 5368709120) as int value in struct literal (overflows)
pkg/object/s3.go:74:29: cannot use 5 << 30 (untyped int constant 5368709120) as int value in struct literal (overflows)
pkg/object/scw.go:47:29: cannot use 5 << 30 (untyped int constant 5368709120) as int value in struct literal (overflows)
pkg/object/tos.go:53:29: cannot use 5 << 30 (untyped int constant 5368709120) as int value in struct literal (overflows)
make: *** [Makefile:31: juicefs.lite] Error 1
``` 

Additional info:
```
juicefs version 1.1.2
OS version machine 1 : Debian GNU/Linux 11 (bullseye)
OS version machine 2: Debian GNU/Linux 12 (bookworm)
kernel machine 1: Linux b3 5.10.177-1 #1 Mon Apr 17 06:42:38 UTC 2023 armv5tel GNU/Linux
kernel machine 2: Linux freedombox 6.1.0-18-armmp-lpae #1 SMP Debian 6.1.76-1 (2024-02-01) armv7l GNU/Linux
```

--- a/pkg/object/interface.go
+++ b/pkg/object/interface.go
@@ -67,7 +67,7 @@ type Limits struct {
 	IsSupportMultipartUpload bool
 	IsSupportUploadPartCopy  bool
 	MinPartSize              int
-	MaxPartSize              int
+	MaxPartSize              int64
 	MaxPartCount             int
 }
 

--- a/pkg/sync/sync.go
+++ b/pkg/sync/sync.go
@@ -449,7 +449,7 @@ func doCopyMultiple(src, dst object.ObjectStorage, key string, size int64, uploa
 		partSize = defaultPartSize
 	}
 	limits := dst.Limits()
-	if size > int64(limits.MaxPartSize)*int64(limits.MaxPartCount) {
+	if size > limits.MaxPartSize*int64(limits.MaxPartCount) {
 		return fmt.Errorf("object size %d is too large to copy", size)
 	}
 	if size > partSize*int64(upload.MaxCount) {
@@ -1051,12 +1051,12 @@ func Sync(src, dst object.ObjectStorage, config *Config) error {
 	if config.Manager != "" {
 		bufferSize = 100
 	}
-	availMem = int64(config.Threads * 32 << 20)
+	availMem = int64(config.Threads) * (32 << 20)
 	tasks := make(chan object.Object, bufferSize)
 	wg := sync.WaitGroup{}
 	concurrent = make(chan int, config.Threads)
 	if config.BWLimit > 0 {
-		bps := float64(config.BWLimit*(1<<20)/8) * 0.85 // 15% overhead
+		bps := float64(int64(config.BWLimit)*(1<<20)/8) * 0.85 // 15% overhead
 		limiter = ratelimit.NewBucketWithRate(bps, int64(bps)*3)
 	}
 

